### PR TITLE
Bunch of nix flake improvements

### PR DESCRIPTION
**Pull request by dpc**
Only after actually landing and enabling Nix Flake based workflow all the smaller and bigger issues resurfaced.

This PR is to get everything in order.

#345 is used to track problems that can addressed later.

### CODE REVIEW

**Comment by dpc** [comment]
OK, so I've got it to the state where `nix-shell` gets a nix flake shell with all binaries built and ready. The problem is that the existing tests want to use binaries from `./target/...` but with Nix the binaries are just in the `$PATH`

File: minimint/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,18 @@ edition = "2018"
 [features]
 telemetry = ["opentelemetry", "opentelemetry-jaeger", "tracing-opentelemetry"]
 
+[lib]
+name = "minimint"
+path = "src/lib.rs"
+
+[[bin]]
+name = "minimint"
+path = "src/bin/server.rs"
```
Comment (justinmoon) [inline minimint/Cargo.toml:18]: Would it be a good idea to rename this file to `minimint.rs` for clarity sake?

**Comment by justinmoon** [comment]
I get the following when I try to run `nix-shell`:

```
error: a 'x86_64-linux' with features {} is required to build '/nix/store/13z0mvkg6cqp9vicy5l5pw2wwibfpbh5-Cargo.toml.drv', but I am a 'aarch64-darwin' with features {benchmark, big-parallel, nixos-test}
```

But `nix develop` works just fine. I'm beginning to doubt whether we should support direct nix usage on mac. Probably best to just support linux and mac users just use a VM or docker.

File: minimint/Cargo.toml (hunk 1 above)
Comment (dpc) [reply minimint/Cargo.toml:18]: Let's do in another PR. There's more changes than I'd like here anyway, and I bet something unexpected is going to need another adjustments then. :)

**Comment by dpc** [comment]
> I get the following when I try to run nix-shell:

Fixed.

Other improvements will go in follow-up commits.

**Review by justinmoon — APPROVED** [review]

**Pull request merged**